Binary files iPhoneAll_old/.DS_Store and iPhoneAll/.DS_Store differ
diff -rupN iPhoneAll_old/AudioUnit.pas iPhoneAll/AudioUnit.pas
--- iPhoneAll_old/AudioUnit.pas	2012-09-10 11:30:34.000000000 +0700
+++ iPhoneAll/AudioUnit.pas	2012-09-10 11:38:52.000000000 +0700
@@ -190,9 +190,9 @@ interface
 
 uses
 {$if defined(IPHONEALL) and not defined(GROUP_INDEPENDENT)}
-	 DefinedClassesCoreMIDI, DefinedClassesAudioUnit, CoreMIDI, AudioToolbox, CoreAudio, CTypes, MacTypes, MacOSXPosix, iPhoneAll;
+	 DefinedClassesCoreMIDI, DefinedClassesAudioUnit, CoreMIDI, CoreAudio, CTypes, MacTypes, MacOSXPosix, iPhoneAll;
 {$else}
-	 DefinedClassesCoreMIDI, DefinedClassesAudioUnit, CoreMIDI, AudioToolbox, CoreFoundation, CoreAudio, CTypes, MacTypes, MacOSXPosix;
+	 DefinedClassesCoreMIDI, DefinedClassesAudioUnit, CoreMIDI, CoreFoundation, CoreAudio, CTypes, MacTypes, MacOSXPosix;
 {$endif}
 
 
diff -rupN iPhoneAll_old/DefinedClassesFoundation.pas iPhoneAll/DefinedClassesFoundation.pas
--- iPhoneAll_old/DefinedClassesFoundation.pas	2012-09-10 11:34:05.000000000 +0700
+++ iPhoneAll/DefinedClassesFoundation.pas	2012-09-10 11:36:05.000000000 +0700
@@ -85,7 +85,7 @@ type
   NSNull = objcclass external;
   NSNumber = objcclass external;
   NSNumberFormatter = objcclass external;
-  NSObject = objcclass external;
+  //NSObject = objcclass external;
   NSOperation = objcclass external;
   NSOperationQueue = objcclass external;
   NSOrderedSet = objcclass external;
@@ -147,7 +147,7 @@ type
   NSMutableCopyingProtocol = objcprotocol external name 'NSMutableCopying';
   NSNetServiceBrowserDelegateProtocol = objcprotocol external name 'NSNetServiceBrowserDelegate';
   NSNetServiceDelegateProtocol = objcprotocol external name 'NSNetServiceDelegate';
-  NSObjectProtocol = objcprotocol external name 'NSObject';
+  //NSObjectProtocol = objcprotocol external name 'NSObject';
   NSPortDelegateProtocol = objcprotocol external name 'NSPortDelegate';
   NSSecureCodingProtocol = objcprotocol external name 'NSSecureCoding';
   NSStreamDelegateProtocol = objcprotocol external name 'NSStreamDelegate';
diff -rupN iPhoneAll_old/audiotoolbox/AudioQueue.inc iPhoneAll/audiotoolbox/AudioQueue.inc
--- iPhoneAll_old/audiotoolbox/AudioQueue.inc	2012-09-10 11:32:58.000000000 +0700
+++ iPhoneAll/audiotoolbox/AudioQueue.inc	2012-09-10 11:40:02.000000000 +0700
@@ -108,14 +108,8 @@ type
     mPacketDescriptionCapacity: UInt32;
     mPacketDescriptions: AudioStreamPacketDescriptionPtr;
     mPacketDescriptionCount: UInt32;
-    {$ifdef __cplusplus}
-    {$endif}
   end;
 type
-  #endif}AudioQueueBuffer = AudioQueueBuffer;
-type
-  #endif}AudioQueueBufferPtr = ^AudioQueueBuffer;
-type
   AudioQueueBufferPtr = ^AudioQueueBuffer;
 
 type
diff -rupN iPhoneAll_old/audiounit/AudioUnitProperties.inc iPhoneAll/audiounit/AudioUnitProperties.inc
--- iPhoneAll_old/audiounit/AudioUnitProperties.inc	2012-09-10 11:32:50.000000000 +0700
+++ iPhoneAll/audiounit/AudioUnitProperties.inc	2012-09-10 11:39:09.000000000 +0700
@@ -65,6 +65,7 @@ const
 
 const
   kAUPresetVersionKey = 'version';
+  kAUPresetTypeKey = 'type';
   kAUPresetSubtypeKey = 'subtype';
   kAUPresetManufacturerKey = 'manufacturer';
   kAUPresetDataKey = 'data';
@@ -571,9 +572,8 @@ const
   kScheduledAudioSliceFlag_BeganToRenderLate = 4;
 
 type
+	ScheduledAudioSlicePtr = ^ScheduledAudioSlice;
   ScheduledAudioSliceCompletionProc = procedure (userData: pointer; bufferList: ScheduledAudioSlicePtr); cdecl;
-
-type
   ScheduledAudioSlice = record
     mTimeStamp: AudioTimeStamp;
     mCompletionProc: ScheduledAudioSliceCompletionProc;
@@ -584,8 +584,6 @@ type
     mNumberFrames: UInt32;
     mBufferList: AudioBufferListPtr;
   end;
-type
-  ScheduledAudioSlicePtr = ^ScheduledAudioSlice;
 
 const
   kAudioUnitProperty_ScheduledFileIDs = 3310;
@@ -595,9 +593,8 @@ const
   kAudioUnitProperty_ScheduledFileNumberBuffers = 3314;
 
 type
+	ScheduledAudioFileRegionPtr = ^ScheduledAudioFileRegion;
   ScheduledAudioFileRegionCompletionProc = procedure (userData: pointer; fileRegion: ScheduledAudioFileRegionPtr; result_: OSStatus); cdecl;
-
-type
   ScheduledAudioFileRegion = record
     mTimeStamp: AudioTimeStamp;
     mCompletionProc: ScheduledAudioFileRegionCompletionProc;
@@ -607,8 +604,6 @@ type
     mStartFrame: SInt64;
     mFramesToPlay: UInt32;
   end;
-type
-  ScheduledAudioFileRegionPtr = ^ScheduledAudioFileRegion;
 {$endif}
 
 {$if  not defined(TARGET_OS_IPHONE)}
diff -rupN iPhoneAll_old/coreaudio/CoreAudioTypes.inc iPhoneAll/coreaudio/CoreAudioTypes.inc
--- iPhoneAll_old/coreaudio/CoreAudioTypes.inc	2012-09-10 11:32:57.000000000 +0700
+++ iPhoneAll/coreaudio/CoreAudioTypes.inc	2012-09-10 11:39:50.000000000 +0700
@@ -69,16 +69,8 @@ type
   AudioBufferList = record
     mNumberBuffers: UInt32;
     mBuffers:array[0..0] of AudioBuffer;
-    {$if (defined(__cplusplus)) and defined(CA_STRICT)}
-    AudioBufferList&): AudioBufferList(;
-    AudioBufferList&)_: AudioBufferList& operator=(;
-    {$endif}
   end;
 type
-  private:AudioBufferList(AudioBufferList&) = AudioBufferList;
-type
-  private:AudioBufferList(AudioBufferList&)Ptr = ^AudioBufferList;
-type
   AudioBufferListPtr = ^AudioBufferList;
 {$endif}
 
@@ -300,16 +292,8 @@ type
     mChannelBitmap: UInt32;
     mNumberChannelDescriptions: UInt32;
     mChannelDescriptions:array[0..0] of AudioChannelDescription;
-    {$if (defined(__cplusplus)) and defined(CA_STRICT)}
-    AudioChannelLayout&): AudioChannelLayout(;
-    AudioChannelLayout&)_: AudioChannelLayout& operator=(;
-    {$endif}
   end;
 type
-  private:AudioChannelLayout(AudioChannelLayout&) = AudioChannelLayout;
-type
-  private:AudioChannelLayout(AudioChannelLayout&)Ptr = ^AudioChannelLayout;
-type
   AudioChannelLayoutPtr = ^AudioChannelLayout;
 
 const
diff -rupN iPhoneAll_old/coredata/NSAtomicStore.inc iPhoneAll/coredata/NSAtomicStore.inc
--- iPhoneAll_old/coredata/NSAtomicStore.inc	2012-09-10 11:33:18.000000000 +0700
+++ iPhoneAll/coredata/NSAtomicStore.inc	2012-09-10 11:36:10.000000000 +0700
@@ -20,7 +20,7 @@ type
     _reserved5: pointer;
   public
     function initWithPersistentStoreCoordinator_configurationName_URL_options (coordinator: NSPersistentStoreCoordinator; configurationName_: NSString; url_: NSURL; options_: NSDictionary): id; message 'initWithPersistentStoreCoordinator:configurationName:URL:options:';
-    function load (error: NSErrorPtr): boolean; message 'load:';
+    function load_ (error: NSErrorPtr): boolean; message 'load:';
     function save (error: NSErrorPtr): boolean; message 'save:';
     function newCacheNodeForManagedObject (managedObject: NSManagedObject): NSAtomicStoreCacheNode; message 'newCacheNodeForManagedObject:';
     procedure updateCacheNode_fromManagedObject (node: NSAtomicStoreCacheNode; managedObject: NSManagedObject); message 'updateCacheNode:fromManagedObject:';
diff -rupN iPhoneAll_old/coredata/NSManagedObjectContext.inc iPhoneAll/coredata/NSManagedObjectContext.inc
--- iPhoneAll_old/coredata/NSManagedObjectContext.inc	2012-09-10 11:33:18.000000000 +0700
+++ iPhoneAll/coredata/NSManagedObjectContext.inc	2012-09-10 11:39:19.000000000 +0700
@@ -24,11 +24,6 @@ var
   NSRefreshedObjectsKey: NSString { available in 10_5, 3_0 }; cvar; external;
   NSInvalidatedObjectsKey: NSString { available in 10_5, 3_0 }; cvar; external;
   NSInvalidatedAllObjectsKey: NSString { available in 10_5, 3_0 }; cvar; external;
-  NSErrorMergePolicy: id { available in 10_4, 3_0 }; cvar; external;
-  NSMergeByPropertyStoreTrumpMergePolicy: id { available in 10_4, 3_0 }; cvar; external;
-  NSMergeByPropertyObjectTrumpMergePolicy: id { available in 10_4, 3_0 }; cvar; external;
-  NSOverwriteMergePolicy: id { available in 10_4, 3_0 }; cvar; external;
-  NSRollbackMergePolicy: id { available in 10_4, 3_0 }; cvar; external;
 {$endif}
 
 {$ifdef TYPES}
diff -rupN iPhoneAll_old/corefoundation/CFBase.inc iPhoneAll/corefoundation/CFBase.inc
--- iPhoneAll_old/corefoundation/CFBase.inc	2012-09-10 11:34:04.000000000 +0700
+++ iPhoneAll/corefoundation/CFBase.inc	2012-09-10 11:36:55.000000000 +0700
@@ -1,90 +1,6 @@
 { Parsed from CoreFoundation.framework (iPhoneOS6.0) CFBase.h }
 { Created on Mon Sep 10 11:34:04 2012 }
 
-{$if  not (defined(__COREFOUNDATION_CFBASE__))}
-
-{$if (defined(_MSC_VER)) and (defined(_M_IX86))}
-{$ifdef TYPES}
-const
-  __i386__ = 1;
-{$endif}
-
-{$endif}
-
-{$if  not (defined(__MACTYPES__))}
-
-{$if  not (defined(_OS_OSTYPES_H))}
-{$ifdef TYPES}
-type
-  UInt8 = char;
-  UInt8Ptr = ^UInt8;
-  SInt8 = char;
-  SInt8Ptr = ^SInt8;
-  UInt16 = cushort;
-  UInt16Ptr = ^UInt16;
-  SInt16 = cshort;
-  SInt16Ptr = ^SInt16;
-  UInt32 = cuint;
-  UInt32Ptr = ^UInt32;
-  SInt32 = cint;
-  SInt32Ptr = ^SInt32;
-  UInt64 = cuint64;
-  UInt64Ptr = ^UInt64;
-  SInt64 = cint64;
-  SInt64Ptr = ^SInt64;
-  OSStatus = SInt32;
-  OSStatusPtr = ^OSStatus;
-{$endif}
-
-{$endif}
-{$ifdef TYPES}
-type
-  Float32 = single;
-  Float32Ptr = ^Float32;
-  Float64 = double;
-  Float64Ptr = ^Float64;
-  UniChar = cushort;
-  UniCharPtr = ^UniChar;
-  UniCharCount = culong;
-  UniCharCountPtr = ^UniCharCount;
-  StringPtr = ^char;
-  ConstStringPtr = ^char;
-  Str255 = array[0..255] of char;
-  Str255Ptr = ^Str255;
-  ConstStr255Param = ^char;
-  OSErr = SInt16;
-  OSErrPtr = ^OSErr;
-  RegionCode = SInt16;
-  RegionCodePtr = ^RegionCode;
-  LangCode = SInt16;
-  LangCodePtr = ^LangCode;
-  ScriptCode = SInt16;
-  ScriptCodePtr = ^ScriptCode;
-  FourCharCode = UInt32;
-  FourCharCodePtr = ^FourCharCode;
-  OSType = FourCharCode;
-  OSTypePtr = ^OSType;
-  Byte = UInt8;
-  BytePtr = ^Byte;
-  SignedByte = SInt8;
-  SignedBytePtr = ^SignedByte;
-{$endif}
-
-{$endif}
-
-{$if  not (defined(__MACTYPES__)) or ((defined(UNIVERSAL_INTERFACES_VERSION)) and (UNIVERSAL_INTERFACES_VERSION < $0340))}
-{$ifdef TYPES}
-type
-  UTF32Char = UInt32;
-  UTF32CharPtr = ^UTF32Char;
-  UTF16Char = UInt16;
-  UTF16CharPtr = ^UTF16Char;
-  UTF8Char = UInt8;
-  UTF8CharPtr = ^UTF8Char;
-{$endif}
-
-{$endif}
-
 {$ifdef EXTERNAL_SYMBOLS}
 var
   kCFCoreFoundationVersionNumber: double; cvar; external;
@@ -210,8 +126,8 @@ type
 {$ifdef TYPES}
 type
   CFTypeRef = pointer;
-  CFStringRef = OpaqueType;
-  CFMutableStringRef = OpaqueType;
+  CFStringRef = CFTypeRef;
+  CFMutableStringRef = CFTypeRef;
   CFPropertyListRef = CFTypeRef;
   CFPropertyListRefPtr = ^CFPropertyListRef;
   CFComparisonResult = CFIndex;
@@ -316,5 +232,3 @@ function CFHash(cf: CFTypeRef): CFHashCo
 function CFCopyDescription(cf: CFTypeRef): CFStringRef; cdecl; external;
 function CFGetAllocator(cf: CFTypeRef): CFAllocatorRef; cdecl; external;
 {$endif}
-
-{$endif}
diff -rupN iPhoneAll_old/corefoundation/CFPlugInCOM.inc iPhoneAll/corefoundation/CFPlugInCOM.inc
--- iPhoneAll_old/corefoundation/CFPlugInCOM.inc	2012-09-10 11:34:04.000000000 +0700
+++ iPhoneAll/corefoundation/CFPlugInCOM.inc	2012-09-10 11:36:27.000000000 +0700
@@ -6,14 +6,6 @@
 const
   __COREFOUNDATION_CFPLUGINCOM__ = 1;
 
-type
-  SInt32 = SInt32;
-  SInt32Ptr = ^SInt32;
-  UInt32 = UInt32;
-  UInt32Ptr = ^UInt32;
-  CFUUIDBytes = CFUUIDBytes;
-  CFUUIDBytesPtr = ^CFUUIDBytes;
-
 const
   SEVERITY_SUCCESS = 0;
   SEVERITY_ERROR = 1;
diff -rupN iPhoneAll_old/coregraphics/Sources.inc iPhoneAll/coregraphics/Sources.inc
--- iPhoneAll_old/coregraphics/Sources.inc	2012-09-10 11:30:32.000000000 +0700
+++ iPhoneAll/coregraphics/Sources.inc	2012-09-10 11:37:55.000000000 +0700
@@ -1,5 +1,6 @@
 {$define FRAMEWORK_LOADED_COREGRAPHICS}
 {$include CoreGraphics.inc}
+{$include CGContext.inc}
 {$include CGGeometry.inc}
 {$include CGAffineTransform.inc}
 {$include CGDataProvider.inc}
@@ -19,7 +20,6 @@
 {$include CGPDFDocument.inc}
 {$include CGFunction.inc}
 {$include CGShading.inc}
-{$include CGContext.inc}
 {$include CGBitmapContext.inc}
 {$include CGDataConsumer.inc}
 {$include CGError.inc}
diff -rupN iPhoneAll_old/coremidi/MIDIDriver.inc iPhoneAll/coremidi/MIDIDriver.inc
--- iPhoneAll_old/coremidi/MIDIDriver.inc	2012-09-10 11:32:49.000000000 +0700
+++ iPhoneAll/coremidi/MIDIDriver.inc	2012-09-10 11:38:40.000000000 +0700
@@ -1,11 +1,6 @@
 { Parsed from CoreMIDI.framework (iPhoneOS6.0) MIDIDriver.h }
 { Created on Mon Sep 10 11:32:49 2012 }
 
-{$ifdef TYPES}
-type
-  MIDIDriverRef = ^MIDIDriverInterfacePtr;
-{$endif}
-
 {$if defined(__LP64__)}
 {$ifdef TYPES}
 type
@@ -22,6 +17,8 @@ type
 {$endif}
 {$ifdef TYPES}
 type
+	MIDIDriverInterfacePtr = ^MIDIDriverInterface;
+	MIDIDriverRef = ^MIDIDriverInterfacePtr;
   MIDIDriverInterface = record
     FindDevices: function (self_: MIDIDriverRef; devList: MIDIDeviceListRef): OSStatus; cdecl;
     Start: function (self_: MIDIDriverRef; devList: MIDIDeviceListRef): OSStatus; cdecl;
@@ -32,8 +29,6 @@ type
     Flush: function (self_: MIDIDriverRef; dest: MIDIEndpointRef; destRefCon1: pointer; destRefCon2: pointer): OSStatus; cdecl;
     Monitor: function (self_: MIDIDriverRef; dest: MIDIEndpointRef; pktlist: MIDIPacketListPtr): OSStatus; cdecl;
   end;
-type
-  MIDIDriverInterfacePtr = ^MIDIDriverInterface;
 {$endif}
 
 {$ifdef EXTERNAL_SYMBOLS}
diff -rupN iPhoneAll_old/coremidi/MIDIServices.inc iPhoneAll/coremidi/MIDIServices.inc
--- iPhoneAll_old/coremidi/MIDIServices.inc	2012-09-10 11:32:49.000000000 +0700
+++ iPhoneAll/coremidi/MIDIServices.inc	2012-09-10 11:38:12.000000000 +0700
@@ -76,26 +76,22 @@ const
   kMIDIInvalidUniqueID = 0;
 
 type
-  MIDINotifyProc = procedure (message_: MIDINotificationPtr; refCon: pointer); cdecl;
+	MIDIPacketPtr = ^MIDIPacket;
+  MIDIPacketListPtr = ^MIDIPacketList;
   MIDIReadProc = procedure (pktlist: MIDIPacketListPtr; readProcRefCon: pointer; srcConnRefCon: pointer); cdecl;
-  MIDICompletionProc = procedure (request: MIDISysexSendRequestPtr); cdecl;
-
-type
   MIDIPacket = record
     timeStamp: MIDITimeStamp;
     length: UInt16;
     data:array[0..255] of Byte;
   end;
-type
-  MIDIPacketPtr = ^MIDIPacket;
-type
   MIDIPacketList = record
     numPackets: UInt32;
     packet:array[0..0] of MIDIPacket;
   end;
+
 type
-  MIDIPacketListPtr = ^MIDIPacketList;
-type
+	MIDISysexSendRequestPtr = ^MIDISysexSendRequest;
+	MIDICompletionProc = procedure (request: MIDISysexSendRequestPtr); cdecl;
   MIDISysexSendRequest = record
     destination: MIDIEndpointRef;
     data: BytePtr;
@@ -105,8 +101,6 @@ type
     completionProc: MIDICompletionProc;
     completionRefCon: pointer;
   end;
-type
-  MIDISysexSendRequestPtr = ^MIDISysexSendRequest;
 
 const
   kMIDIMsgSetupChanged = 1;
@@ -122,12 +116,13 @@ type
   MIDINotificationMessageIDPtr = ^MIDINotificationMessageID;
 
 type
+	MIDINotificationPtr = ^MIDINotification;
+	MIDINotifyProc = procedure (message_: MIDINotificationPtr; refCon: pointer); cdecl;
   MIDINotification = record
     messageID: MIDINotificationMessageID;
     messageSize: UInt32;
   end;
-type
-  MIDINotificationPtr = ^MIDINotification;
+
 type
   MIDIObjectAddRemoveNotification = record
     messageID: MIDINotificationMessageID;
diff -rupN iPhoneAll_old/foundation/NSObject.inc iPhoneAll/foundation/NSObject.inc
--- iPhoneAll_old/foundation/NSObject.inc	2012-09-10 11:34:08.000000000 +0700
+++ iPhoneAll/foundation/NSObject.inc	2012-09-10 11:35:41.000000000 +0700
@@ -20,7 +20,7 @@ type
 {$endif}
 
 {$ifdef PROTOCOLS}
-
+(*
 type
   NSObjectProtocol = objcprotocol external name 'NSObject'
     function isEqual (object_: id): boolean; message 'isEqual:';
@@ -45,7 +45,7 @@ type
   optional
     function debugDescription: NSString; message 'debugDescription';
   end;
-
+*)
 type
   NSCopyingProtocol = objcprotocol external name 'NSCopying'
     function copyWithZone (zone: NSZonePtr): id; message 'copyWithZone:';
@@ -70,7 +70,7 @@ type
 {$endif}
 
 {$ifdef CLASSES}
-
+(*
 type
   NSObject = objcclass external (NSObjectProtocol)
   private
@@ -128,7 +128,7 @@ type
     function superclass: pobjc_class; message 'superclass';
     function zone: NSZonePtr; message 'zone';
   end;
-
+*)
 
 type
   NSCoderMethods = objccategory external (NSObject)
diff -rupN iPhoneAll_old/foundation/NSZone.inc iPhoneAll/foundation/NSZone.inc
--- iPhoneAll_old/foundation/NSZone.inc	2012-09-10 11:34:08.000000000 +0700
+++ iPhoneAll/foundation/NSZone.inc	2012-09-10 13:55:52.000000000 +0700
@@ -2,9 +2,9 @@
 { Created on Mon Sep 10 11:34:08 2012 }
 
 {$ifdef TYPES}
-type
-  NSZone = OpaqueType;
-  NSZonePtr = ^NSZone;
+//type
+//  NSZone = OpaqueType;
+//  NSZonePtr = ^NSZone;
 {$endif}
 
 {$ifdef FUNCTIONS}
diff -rupN iPhoneAll_old/uikit/UIBezierPath.inc iPhoneAll/uikit/UIBezierPath.inc
--- iPhoneAll_old/uikit/UIBezierPath.inc	2012-09-10 11:34:11.000000000 +0700
+++ iPhoneAll/uikit/UIBezierPath.inc	2012-09-10 11:39:27.000000000 +0700
@@ -17,7 +17,7 @@ const
   UIRectCornerTopRight = 1 shl 1;
   UIRectCornerBottomLeft = 1 shl 2;
   UIRectCornerBottomRight = 1 shl 3;
-  UIRectCornerAllCorners = not(0)UL;
+  UIRectCornerAllCorners = 0;
 {$endif}
 
 {$ifdef CLASSES}
